src/core/eval.ml

```diff
@@ -470,42 +496,49 @@ let hnf =
   Debug.(record_time Rewriting (fun () -> r := hnf c t)); !r
 
 (** [eq_modulo c a b] tests the convertibility of [a] and [b] in context
-   [c]. WARNING: may have side effects in TRef's introduced by whnf. *)
+    [c]. WARNING: may have side effects in TRef's introduced by whnf. *)
 let eq_modulo : ctxt -> term -> term -> bool = fun c ->
-  eq_modulo whnf (cfg_of_ctx c true)
+  eq_modulo whnf (Config.make c)
 
 let eq_modulo =
   let open Stdlib in let r = ref false in fun c t u ->
   Debug.(record_time Rewriting (fun () -> r := eq_modulo c t u)); !r
 
 (** [pure_eq_modulo c a b] tests the convertibility of [a] and [b] in context
-   [c] with no side effects. *)
+    [c] with no side effects. *)
 let pure_eq_modulo : ctxt -> term -> term -> bool = fun c a b ->
   Timed.pure_test (fun (c,a,b) -> eq_modulo c a b) (c,a,b)
 
 (** [whnf c t] computes a whnf of [t], unfolding the variables defined in the
    context [c], and using user-defined rewrite rules if [~rewrite]. *)
-let whnf : ?rewrite:bool -> ctxt -> term -> term = fun ?(rewrite=true) c t ->
+let whnf :
+     ?problem:problem
+  -> ?rewrite:rw_tag list
+  -> ctxt -> term -> term =
+  fun ?problem ?rewrite c t ->
   Stdlib.(steps := 0);
-  let u = whnf (cfg_of_ctx c rewrite) t in
+  let u = whnf (Config.make ?problem ?rewrite c) t in
   let r = if Stdlib.(!steps = 0) then unfold t else u in
   (*if Logger.log_enabled () then
     log_eval "whnf %a%a\n= %a" pp_ctxt c pp_term t pp_term r;*) r
 
 let whnf =
-  let open Stdlib in let r = ref mk_Kind in fun ?rewrite c t ->
-  Debug.(record_time Rewriting (fun () -> r := whnf ?rewrite c t)); !r
+  let open Stdlib in let r = ref mk_Kind in fun ?problem ?rewrite c t ->
+  Debug.(
+    record_time Rewriting (fun () -> r := whnf ?problem ?rewrite c t));
+  !r
 
 (** [simplify t] computes a beta whnf of [t] belonging to the set S such that:
 - terms of S are in beta whnf normal format
 - if [t] is a product, then both its domain and codomain are in S. *)
 let rec simplify : term -> term = fun t ->
-  match get_args (whnf ~rewrite:false [] t) with
+  let rewrite = [`NoRw; `NoExpand ] in
```

Comment:
extra space before ]

Reply:
We should use automatic formatting for that kind of stuff